js/src/tags.js

```diff
@@ -0,0 +1,467 @@
+            this.errors_banner.show();
+        } else {
+            this.errors_banner.hide();
+        }
+
+        var displayed_ids = this.displayed_ids;
+
+        if (displayed_ids.length === 0) {
+            this.no_results_banner.show();
+        } else {
+            this.no_results_banner.hide();
+            for (var offset in displayed_ids) {
+                var item = this.getItem(displayed_ids[offset]);
+                this.items_list.append(this.displayRow(displayed_ids[offset], item));
+            }
+        }
+
+        this.checkModified();
+    };
+
+    ListableItemsAdmin.prototype.displayRow = function (id, item) {
+        var new_item = this.new_item_ids.includes(id);
+        var edit_mode = (id in this.unlocked_items);
+        var modified = (id in this.modified_items);
+        var deleted = this.deleted_items_ids.includes(id);
+
+        var template = $($.parseHTML(_.template(this.html_template.html())({})));
+
+        if (new_item) {
+            template.addClass(this.row_new_class);
+        } else if (deleted) {
+            template.addClass(this.row_deleted_class);
+        } else if (modified) {
+            template.addClass(this.row_modified_class);
+        }
+
+        var edit_button = template.find('.' + this.button_class_edit);
+        var reset_button = template.find('.' + this.button_class_reset);
+        var delete_button = template.find('.' + this.button_class_delete);
+
+        template.find('.' + this.class_editable_field).prop('disabled', !edit_mode);
+
+        edit_button.text(edit_mode ? this.localised_save_edit : this.localised_edit);
+        delete_button.text(this.localised_delete);
+        reset_button.text(this.localised_reset);
+
+        edit_button.hide();
+        reset_button.hide();
+        delete_button.hide();
+
+        if ((!new_item && (deleted || modified)) || (new_item && edit_mode)) {
+            reset_button.show();
+        }
+        if (!deleted) {
+            edit_button.show();
+        }
+        if ((new_item && !edit_mode) || (!deleted && !edit_mode && !modified)) {
+            delete_button.show();
+        }
+
+        template.attr('data-id', id);
+
+        this.fillRowFields(id, template, item);
+
+        return template;
+    };
+
+    ListableItemsAdmin.prototype.fillRowFields = function (id, template, item) {
+        var self = this;
+        template.find('.' + this.class_editable_field).each(function () {
+            var field = $(this);
+            var type = field.attr('type');
+            var name = field.attr('data-name');
+            //TODO: Add lodash 'has' for deeply complex objects, once lodash exists.
+            var value = (name in item ? item[name] : null);
+
+            if (type === 'text') {
+                field.val(value);
+            } else if (type === 'checkbox') {
+                field.attr('checked', value);
+            } else {
+                console.log('Unsupported field type ' + type);
```

Comment:
`console.warn` would be clearer

Reply:
Fixed in ef1176accf5c411a39af7103186807ffe773720a